frontend/src/pages/UserDetails.tsx

```diff
@@ -150,7 +150,27 @@ const UserDetailsPage: React.FC = () => {
               </div>
             </div>
             <div className="px-6 py-6">
-              {user.bio && <p className="text-lg text-gray-700 dark:text-gray-300">{user.bio}</p>}
+              {user.bio && (
+                <p className="text-lg text-gray-700 dark:text-gray-300">
+                  {user.bio.split(' ').map((word, index) => {
+                    const mentionMatch = word.match(/^@(\w+)/)
```

Comment:
This works for the link but it still shows the dot highlighted in the UI. You should be able to fix that by updating both regexes here and at the top. The taking out the correct group item. 
Also, could we move this logic out of the template? It's getting messy.
![Screenshot 2025-03-02 at 9 56 30 AM](https://github.com/user-attachments/assets/303a4778-a5d0-4ab5-bd5c-34d15d174c77)